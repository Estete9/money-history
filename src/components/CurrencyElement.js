import PropTypes from 'prop-types';
import styles from '../styles/currencyElement.module.css';

function CurrencyElement({ currency }) {
  return (
    <div className={styles.td}>
      <img src="/" alt="currencyImg.name" />
      <div className={styles.currencyContent}>
        <p>{`currency name (needs to change later) ${currency}`}</p>
        <p>{`currency symbol ${currency}`}</p>
      </div>
    </div>
  );
}

CurrencyElement.propTypes = {
  currency: PropTypes.shape({
    icon: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    symbol: PropTypes.string.isRequired,
  }),
};

CurrencyElement.defaultProps = {
  currency: {
    icon: '',
    name: '',
    symbol: '',
  },
};

export default CurrencyElement;
