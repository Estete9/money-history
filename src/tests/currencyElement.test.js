import React from 'react';
import renderer from 'react-test-renderer';
import CurrencyElement from '../components/CurrencyElement';

test('Checks that CurrencyElement component renders correctly', () => {
  const currency = {
    currencyCountry: 'Australian Dollar',
    currencySymbol: 'AUD',
  };
  const component = renderer.create(<CurrencyElement currency={currency} />);
  const currencyElementTree = component.toJSON();
  expect(currencyElementTree).toMatchSnapshot();
});
