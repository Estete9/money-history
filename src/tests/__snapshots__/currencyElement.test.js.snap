// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks that CurrencyElement component renders correctly 1`] = `
<div
  className="td"
>
  <div
    className="enterArrowWrapper"
  >
    <img
      alt="enter currency button"
      className="enterCurrencyArrow"
      src="svg-mock"
    />
  </div>
  <img
    alt="currencyImg"
    className="currencyElementImg"
  />
  <div
    className="currencyContent"
  >
    <p>
      Australian Dollar
    </p>
    <p>
      AUD
    </p>
  </div>
</div>
`;
